galsim/config/input_powerspectrum.py

```diff
@@ -124,25 +189,41 @@ def _GenerateFromPowerSpectrumShear(config, base, value_type):
     # one present.
     galsim.config.CheckAllParams(config, opt={ 'num' : int })
 
-    try:
+    with warnings.catch_warnings(record=True) as w:
         g1,g2 = power_spectrum.getShear(pos)
+    if len(w) > 0:
+        # Send the warning to the logger, rather than raising a normal warning.
+        # The warning here would typically be that the position is out of range of where the
+        # power spectrum is defined.  So if we do get this and the position is not on the image,
+        # we probably don't care.  In that case, just log it as debug, now warn.
+        log_level = (logging.WARNING if 'current_image' in base and
+                                        base['current_image'].getOuterBounds().includes(pos)
+                     else logging.DEBUG)
+        for ww in w:
+            logger.log(log_level, 'obj %d: %s',base['obj_num'], ww.message)
+
+    try:
         shear = galsim.Shear(g1=g1,g2=g2)
     except KeyboardInterrupt:
         raise
     except Exception as e:
-        import warnings
-        warnings.warn("Warning: PowerSpectrum shear is invalid -- probably strong lensing!  " +
-                      "Using shear = 0.")
+        if logger:
+            logger.warning('obj %d: Warning: PowerSpectrum shear (g1=%f, g2=%f) is invalid. '%(
```

Comment:
Actually, I think this behavior goes counter to how we deal with such things elsewhere in the config layer.  We only send information to a logger object.  So if the user wants to get these kinds of messages, they provide a logger.  Otherwise, it should be silent about this.  So I switched it to just use the `logger.warning` call.   b2fbbc917